mcp/src/main/java/io/airlift/mcp/model/CallToolRequest.java

```diff
@@ -0,0 +1,23 @@
+package io.airlift.mcp.model;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public record CallToolRequest(String name, Map<String, Object> arguments, Optional<Map<String, Object>> meta)
```

Comment:
nit: metadata?

Reply:
The names here are fixed as they are part of the spec. Anything in the `model` package is fixed.